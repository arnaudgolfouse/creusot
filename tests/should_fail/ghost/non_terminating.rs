extern crate creusot_contracts;
use creusot_contracts::*;

#[check(ghost)]
pub fn recursive() {
    ghost! {
        f();
        42
    };
}

#[check(ghost)]
fn f() {
    recursive();
}

#[allow(unreachable_code)]
pub fn looping() {
    let _g: Ghost<i32> = ghost! {
        loop {}
    };
}
