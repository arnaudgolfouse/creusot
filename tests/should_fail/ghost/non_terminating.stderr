error: `ghost!` blocks must not contain loops.
  --> non_terminating.rs:20:9
   |
20 |         loop {}
   |         ^^^^^^^

error: Mutually recursive functions: when calling `f`...
  --> non_terminating.rs:13:1
   |
13 | fn f() {
   | ^^^^^^
   |
note: then `f` calls `recursive`...
  --> non_terminating.rs:14:5
   |
14 |     recursive();
   |     ^^^^^^^^^^^
note: finally `recursive` calls `f`.
  --> non_terminating.rs:7:9
   |
 7 |         f();
   |         ^^^

error: aborting due to 2 previous errors

